### DISCUSSION TITLE
new api

### DISCUSSION THREAD

**Discussion by casualmatt** [postalserver/postal#1977] [Development discussions]
Hello, 

In the discussion last year about, postal announced a new API interface, after the update on there update to [apia](https://github.com/krystal/apia) there API framework.

There is any update on those new endpoints? 

btw Postal is really good, it just miss those API to be an alternative to Proxmox mail gateway. 

Matt.

**Comment by willpower232** [comment]
I think Postal is still using [moonrope](https://github.com/adamcooke/moonrope) so I don't think theres been any changes to Postal.

Its also worth saying that Proxmox Mail Gateway is a focused on email security whereas Postal is focused on sending emails so there may not be as much overlap as you want.

**Reply by casualmatt** [reply]
my comment was related in fact to the statement on the Postal v2 update. 

> New HTTP API. The current API provides a very small number of features. We have a new API framework that we will be open sourcing in the next few months and we want to use it to provide a far more feature-rich HTTP API. Our new framework generate [beautiful documentation](https://developers.katapult.io/api) and would be an excellent addition to Postal.

[Quote ](https://github.com/postalserver/postal/discussions/1513)

PMG, is a routing/filtering software, but for what is regarding AntiSpam, due to the use of RspamD, Postal is in fact the most effective solution, and the feature like tracking are really a good plus.

**Comment by pliashkou** [comment]
Hi guys, were you able to integrate Apia?

**Comment by FreeMinded** [comment]
Hi, we have been looking into Postal and it seems perfect for our use case but we would need an API to create and manage domains and users?
Has there been any progress on the v2 API mentioned in the [docs](https://docs.postalserver.io/developer/api)? 
We have Ruby on Rails Devs inhouse and would be able and willing to contribute to the API development if there is support from you guys. @willpower232 @adamcooke

**Comment by csiocc** [comment]
Because no one has responded to our requests, we have created our own fork: [Postal-api-extended](https://github.com/csiocc/postal-api-extended) We will maintain it until Postal API V2 is released, and it covers our current use case.

@willpower232 @adamcooke We are still happy to contribute to the new API.

**Reply by gregorygibbons52-afk** [reply]
Does this API you created offer the missing API management features we have been waiting for in the ellusive Postal API V2?  Im using Mailcow because current Postal API doesnt allow for managenment features. I felt more home, otherwise, in Postal.

**Comment by adamcooke** [comment]
Good news! I have now been able to start work on this. Things are looking good but I do need to spend some more time putting more tests in place before I'll be confident releasing this. However, things are looking good for API v2.

* REST-like API
* New API tokens
  * Tokens can be global, at the org level or at the server level
  * Scopes control which functionality (area + read and/or write) they can use
  * Tokens have an expiry option
* All functionality is available through the API.
* An API explorer is available in the UI to act as authoritative documentation. 
* JSON-based API schema available to aid with automatic generation of API clients.
 
<img width="2636" height="2724" alt="image" src="https://github.com/user-attachments/assets/ed57bfe3-e1ab-4490-b6ce-fbcb5b9da07f" />

<img width="2678" height="3266" alt="image" src="https://github.com/user-attachments/assets/2dd5b946-c2d1-47f0-8a90-68e801fbb041" />

This will likely be part of the 3.4 release. I'll get things pushed up to a branch shortly.

**Reply by uhlhosting** [reply]
@adamcooke great job! Keep it up! Id say the sooner this goes live, the sooner the community can jump in and improve or at least complain about it. Either way, this is much welcomed, and long awaited feature. Keep it up!

**Reply by csiocc** [reply]
Nice!

**Comment by Jaq112** [comment]
any updates on v2? or an idea of release date?
